src/metaswitch/clearwater/plugin_tests/test_config_manager_plugin.py

```diff
@@ -0,0 +1,107 @@
+# the program along with SSL, set forth below. This program is distributed
+# in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details. You should have received a copy of the GNU General Public
+# License along with this program.  If not, see
+# <http://www.gnu.org/licenses/>.
+#
+# The author can be reached by email at clearwater@metaswitch.com or by
+# post at Metaswitch Networks Ltd, 100 Church St, Enfield EN2 6BQ, UK
+#
+# Special Exception
+# Metaswitch Networks Ltd  grants you permission to copy, modify,
+# propagate, and distribute a work formed by combining OpenSSL with The
+# Software, or a work derivative of such a combination, even if such
+# copying, modification, propagation, or distribution would otherwise
+# violate the terms of the GPL. You must comply with the GPL in all
+# respects for all of the code used other than OpenSSL.
+# "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+# Project and licensed under the OpenSSL Licenses, or a work based on such
+# software and licensed under the OpenSSL Licenses.
+# "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+# under which the OpenSSL Project distributes the OpenSSL toolkit software,
+# as those licenses appear in the file LICENSE-OPENSSL.
+
+import unittest
+import mock
+import logging
+import uuid
+
+_log = logging.getLogger()
+
+from clearwater_etcd_plugins.clearwater_config_manager.shared_config_plugin import SharedConfigPlugin
+from metaswitch.clearwater.cluster_manager.plugin_base import PluginParams
+
+
+class TestConfigManagerPlugin(unittest.TestCase):
+    @mock.patch('clearwater_etcd_plugins.clearwater_config_manager.shared_config_plugin.safely_write')
+    @mock.patch('clearwater_etcd_plugins.clearwater_config_manager.shared_config_plugin.run_command')
+    def test_config_changed(self, mock_run_command, mock_safely_write):
+        """Test Config Manager writes new config when config has changed"""
+
+        # Create the plugin
+        plugin = SharedConfigPlugin(PluginParams(ip='10.0.0.1',
+                                                 mgmt_ip='10.0.1.1',
+                                                 local_site='local_site',
+                                                 remote_site='remote_site',
+                                                 remote_cassandra_seeds='',
+                                                 signaling_namespace='',
+                                                 uuid=uuid.UUID('92a674aa-a64b-4549-b150-596fd466923f'),
+                                                 etcd_key='etcd_key',
+                                                 etcd_cluster_key='etcd_cluster_key'))
+
+        # Set up the config strings to be tested
+        old_config_string = "Test config string here. \n Replace with replica real config soon"
```

Comment:
Are you going to change this test string? I don't think you need to make the test string more like the shared config (but I would like string to make sense)